src/lit-html.ts

```diff
@@ -12,846 +12,156 @@
-    templateCache.set(result.strings, template);
-  }
-  return template;
-}
-
-export type TemplateContainer = (Element|DocumentFragment)&{
-  __templateInstance?: TemplateInstance;
-};
-
-/**
- * Renders a template to a container.
+ * Properties are set by prefixing an attribute name with `.`.
  *
- * To update a container with new values, reevaluate the template literal and
- * call `render` with the new result.
+ * Attribute names in lit-html templates preserve case, so properties are case
+ * sensitive. If an
+ * expression takes up an entire attribute value, then the property is set to
+ * that value. If an expression is interpolated with a string or other
+ * expressions then the property is set to the string result of the
+ * interpolation.
  *
- * @param result a TemplateResult created by evaluating a template tag like
- *     `html` or `svg`.
- * @param container A DOM parent to render to. The entire contents are either
- *     replaced, or efficiently updated if the same result type was previous
- *     rendered there.
- * @param templateFactory a function to create a Template or retreive one from
- *     cache.
- */
-export function render(
-    result: TemplateResult,
-    container: Element|DocumentFragment,
-    templateFactory: TemplateFactory = defaultTemplateFactory) {
-  const template = templateFactory(result);
-  let instance = (container as TemplateContainer).__templateInstance;
-
-  // Repeat render, just call update()
-  if (instance !== undefined && instance.template === template &&
-      instance._partCallback === result.partCallback) {
-    instance.update(result.values);
-    return;
-  }
-
-  // First render, create a new TemplateInstance and append it
-  instance =
-      new TemplateInstance(template, result.partCallback, templateFactory);
-  (container as TemplateContainer).__templateInstance = instance;
-
-  const fragment = instance._clone();
-
-  removeNodes(container, container.firstChild);
-  container.appendChild(fragment);
-
-  instance.update(result.values);
-}
-
-/**
- * An expression marker with embedded unique key to avoid collision with
- * possible text in templates.
- */
-const marker = `{{lit-${String(Math.random()).slice(2)}}}`;
-
-/**
- * An expression marker used text-positions, not attribute positions,
- * in template.
- */
-const nodeMarker = `<!--${marker}-->`;
-
-const markerRegex = new RegExp(`${marker}|${nodeMarker}`);
-
-/**
- * This regex extracts the attribute name preceding an attribute-position
- * expression. It does this by matching the syntax allowed for attributes
- * against the string literal directly preceding the expression, assuming that
- * the expression is in an attribute-value position.
+ * Example:
  *
- * See attributes in the HTML spec:
- * https://www.w3.org/TR/html5/syntax.html#attributes-0
+ *     html`<input .value=${value}>`
  *
- * "\0-\x1F\x7F-\x9F" are Unicode control characters
+ * Event handlers are sset by prefixing an attribute name with `on-`.
```

Comment:
typo: sset => set

Change `on-` to `@` right?

Reply:
done